packages/cmr-client/src/EarthdataToken.ts

```diff
@@ -0,0 +1,183 @@
+// @ts-nocheck
+import get from 'lodash/get';
+import got from 'got';
+import Response from 'got';
+
+type EarthdataGetTokenResponse = Response<{
+  body: {
+    access_token?: string,
+    token_type?: string,
+    expiration_date?: string
+  }
+}>;
+
+type EarthdataPostTokenResponse = Response<{
+  body: {
+    access_token?: string,
+    expiration_date?: string
+  }
+}>;
+
+export interface EarthdataTokenParams {
+  username: string,
+  password: string,
+  edlEnv: string,
+  token?: string,
+}
+
+/**
+ * A class to simplify requests for the Earthdata Login token used for CMR
+ *
+ * @typicalname earthdataToken
+ *
+ * @example
+ * const { EarthdataToken } = require('@cumulus/cmr-client');
+ *
+ * const earthdataToken = new EarthdataToken({
+ *  "username": "my-username",
+ *  "password": "my-password",
+ *  "edlEnv": "my-cmr-environment",
+ *  "token" : "my-token"
+ * });
+ */
+export class EarthdataToken {
+  username: string;
+  password: string;
+  edlEnv: string;
+  token?: string;
```

Comment:
Is there use case you need to pass in the token? If not, remove from attribute list.